cmd/hegel/main.go

```diff
@@ -24,7 +30,183 @@ var (
+		return nil, err
+	}
+
+	return rootCmd, nil
+}
+
+// Run executes rc. It satisfies cobra.Command's RunE field.
+func (rc *RootCommand) Run(*cobra.Command, []string) error {
+	return nil
+}
+
+func configureAllFlags(vpr *viper.Viper, set *pflag.FlagSet) error {
+	if err := configureFlags(vpr, set); err != nil {
+		return err
+	}
+
+	return configureLegacyFlags(vpr, set)
+}
+
+func configureFlags(vpr *viper.Viper, set *pflag.FlagSet) error {
+	set.String("facility", "onprem", "The facility we are running in (mostly to connect to cacher)")
+	set.String("data-model", datamodel.TinkServer, "The back-end data source: [\"1\", \"kubernetes\"] (1 indicates tink server)")
+	set.String("trusted-proxies", "", "A commma separated list of allowed peer IPs and/or CIDR blocks to replace with X-Forwarded-For for both gRPC and HTTP endpoints")
+
+	set.String("grpc-tls-cert", "", "Path of a TLS certificate for the gRPC server")
+	set.String("grpc-tls-key", "", "Path to the private key for the tls_cert")
+	set.Bool("grpc-use-tls", true, "Toggle for gRPC TLS usage")
+
+	set.Int("http-port", 50061, "Port to listen on for HTTP requests")
+	set.String("http-custom-endpoints", `{"/metadata":".metadata.instance"}`, "JSON encoded object specifying custom endpoint => metadata mappings")
+
+	set.String("kubernetes-api", "", "URL of the Kubernetes API Server")
+	set.String("kubeconfig", "", "Path to a kubeconfig file")
+
+	if err := vpr.BindPFlags(set); err != nil {
+		return err
+	}
+
+	var err error
+	set.VisitAll(func(f *pflag.Flag) {
+		if err != nil {
+			return
+		}
+		err = vpr.BindEnv(f.Name)
+	})
+
+	return err
+}
+
+func configureLegacyFlags(vpr *viper.Viper, set *pflag.FlagSet) error {
+	set.SetNormalizeFunc(func(f *pflag.FlagSet, name string) pflag.NormalizedName {
+		switch name {
+		case "use_tls":
+			return pflag.NormalizedName("grpc-use-tls")
+		case "tls_cert":
+			return pflag.NormalizedName("grpc-tls-cert")
+		case "tls_key":
+			return pflag.NormalizedName("grpc-tls-key")
+		case "http_port":
+			return pflag.NormalizedName("http-port")
+		default:
+			return pflag.NormalizedName(name)
+		}
+	})
+
+	for key, envName := range map[string]string{
+		"data-model":            "DATA_MODEL_VERSION",
+		"grpc-tls-cert":         "HEGEL_TLS_CERT",
+		"grpc-tls-key":          "HEGEL_TLS_KEY",
+		"grpc-use-tls":          "USE_TLS",
+		"http-custom-endpoints": "CUSTOM_ENDPOINTS",
+		"trusted-proxies":       "TRUSTED_PROXIES",
+	} {
+		if err := vpr.BindEnv(key, envName); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// HEGEL_FACILITY
+// HEGEL_HTTP_PORT
+// HEGEL_TLS_CERT
+// HEGEL_TLS_KEY
+// HEGEL_USE_TLS
+// DATA_MODEL_VERSION
+// HEGEL_KUBE_API
+// HEGEL_KUBECONFIG
+// CUSTOM_ENDPOINTS
+// TRUSTED_PROXIES
+
+func handleMainError(err error) {
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v", err)
+		os.Exit(1)
+	}
+}
+
 func main() {
+	cmd, err := NewRootCommand()
+	handleMainError(err)
+	handleMainError(cmd.Execute())
+	os.Exit(0)
```

Comment:
Why exit here?

Reply:
Ignore it, its temporary. This was just for testing purposes to make sure the CLI worked correctly with env vars etc.

I'll probably strip this out and rework the sources on this PR, then introduce the actual changes on a separate PR to keep things isolated and easier to review.